StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1412CodeFixProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace StyleCop.Analyzers.MaintainabilityRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.Text;
+
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1412StoreFilesAsUtf8"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>To fix a violation of this rule, change the encoding to utf-8 with preamble.</para>
+    /// </remarks>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1412CodeFixProvider))]
+    [Shared]
+    public class SA1412CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1412StoreFilesAsUtf8.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
```

Comment:
:exclamation: This batch fixer won't work when adding files. You'll need to either return `null` or implement a custom fix all provider for this.